geoh5py/io/h5_reader.py

```diff
@@ -192,6 +192,9 @@ def fetch_concatenated_values(
                 if attribute.dtype in [float, "float64", "float32"]:
                     attribute[attribute == FLOAT_NDV] = np.nan
 
+                if attribute.dtype == object and isinstance(attribute[0], bytes):
```

Comment:
I don't know if it's pertinent, but to ensure no future error, we could check the type here and raise an error?

Reply:
Well technically the format should be correct since it comes from geoh5. Not sure I am following.

Reply:
Ok then not type testing